src/bootstrap/src/core/config/config.rs

```diff
@@ -1218,11 +1218,10 @@ impl Config {
                     exit!(2);
                 })
         }
-        Self::parse_inner(args, get_toml)
+        Self::parse_inner(flags, get_toml)
     }
 
-    pub(crate) fn parse_inner(args: &[String], get_toml: impl Fn(&Path) -> TomlConfig) -> Config {
-        let mut flags = Flags::parse(args);
+    pub(crate) fn parse_inner(mut flags: Flags, get_toml: impl Fn(&Path) -> TomlConfig) -> Config {
```

Comment:
Q: Why is `flags` being mutable here?

Reply:
This was pre-existing. A few lines below the code does `mem::take(&mut flags.paths)` (no idea why).